feat: add password reset process pages

Maaayhan opened the pull request:
Resolves CP-28 & CP-121

[CP-28](https://beequant.atlassian.net/browse/CP-28) add forgot password process
[CP-121](https://beequant.atlassian.net/browse/CP-121) turbo migration

---

## What happens here?

- Implemented "Forgot Password" and "Reset Password" forms
- Created a Success Reset page to confirm successful password reset
- Added appropriate user guidance messages throughout the reset process
- Updated AuthFormContainer for reuse

### UI comparison
**Legacy:**
<img width="439" alt="image" src="https://github.com/user-attachments/assets/94f8b7da-6fd9-457e-bf56-2a9ffb8c7731">

**Turbo:**
<img width="540" alt="image" src="https://github.com/user-attachments/assets/2bfe7b99-04dc-4b06-97f2-a28b34c6b42c">

**Legacy:**
<img width="540" alt="image" src="https://github.com/user-attachments/assets/bcd5c0a6-1b3f-4be1-8395-123abaa19fe7">

**Turbo:**
<img width="600" alt="image" src="https://github.com/user-attachments/assets/c2405e9f-1be4-4b80-a511-707cd67e9553">

**Legacy:**
<img width="539" alt="image" src="https://github.com/user-attachments/assets/4f5e55e5-aa1d-43ff-83b1-6cbf44e1a841">

**Turbo:**
<img width="546" alt="image" src="https://github.com/user-attachments/assets/e5fbe4ac-d4dd-469b-a3e0-f6dcbd84cc79">


## How do I know this is working?
- For testing purposes, please switch the backend to the branch ‘**feat/CP28-reset-user-password**’ in platform_api repo.
- When the user clicks the "Forgot password" text on the login page, they are directed to the Reset Password Initiation Page, which will prompt them to enter their registered email address.
![image](https://github.com/user-attachments/assets/88222198-5e9a-4172-8a81-2c9fa16660e4)
- After submission, the system will display the message "If the email is associated with an account, a reset email will be sent shortly. Please check your mailbox," guiding the user to check their email for the reset link. If an error occurs during the request, the message "An error occurred while processing your request. Please try again later." will be shown.
![image](https://github.com/user-attachments/assets/38a0eb43-461c-4ec5-9681-2ee13e016157)

- Clicking the link in the email will navigate the user to the Reset Password Page, where they can enter a new password.
![image](https://github.com/user-attachments/assets/e83a2428-ec50-4401-a28f-f8ad305956c3)

- If an error occur during the process, the error message will be shown.
![image](https://github.com/user-attachments/assets/7d863c79-db62-43ca-a9e6-0ca7fd6151fb)

- If the reset is successful, the user will be navigated to the reset success page. On this page, the user can directly click the "Back to Login" button to log in on the Login Page.
![image](https://github.com/user-attachments/assets/1cd114e7-35a9-45db-9d6b-b9eb5508ca34)

- During the process, users can always click the "Login" text at the bottom to return to the login page.
<img width="529" alt="image" src="https://github.com/user-attachments/assets/d640bba7-a63c-4cbd-8d4b-f382d4366e9b">

## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->

Chris-Wang reviewed (changes requested):
Need resolve conflict in this PR

Maaayhan committed:
feat: add password reset process pages

- Implemented "Forgot Password" and "Reset Password" forms
- Created a Success Reset page to confirm successful password reset
- Added appropriate user guidance messages throughout the reset process
- Updated AuthFormContainer for reuse

Resolves CP-28 / CP-121

Chris-Wang reviewed (approved):
LGTM
